pkg/controller/windowsmachine/nodeconfig/nodeconfig.go

```diff
@@ -129,10 +132,16 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.setNode(); err != nil {
 		return errors.Wrapf(err, "error getting node object for VM %s", nc.ID())
 	}
-	// Apply worker labels
-	if err := nc.applyWorkerLabel(); err != nil {
-		return errors.Wrap(err, "failed applying worker label")
+	// Apply labels and annotations to nc.node
+	nc.addWorkerLabel()
+	nc.addVersionAnnotation()
+	// update node object
+	node, err := nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, metav1.UpdateOptions{})
+	if err != nil {
```

Comment:
Should we fail immediately or retry a few times? It is very possible that a change could be in flight when this update is made.

Reply:
Its most likely safer to just requeue on error